### ISSUE TITLE
Install app error with greenlet installation

### ISSUE THREAD

**Issue by rastaferraille** [PnX-SI/GeoNature-atlas#494]
Hello,

I'm trying to install geonature atlas on a debian 12.1 server.
There is an error using install_app.sh script (or using pip install -r requirements.txt).

src/greenlet/greenlet.h:45:5: error: unknown type name `CFrame`

error: legacy-install-failure

I know that 12.1 version of Debian has not been tested, but we can't user older versions for security reasons.

Thank you for your help !

Etienne

**Comment by camillemonchicourt** [comment]
Yes, we couldn't spend time yet to make GeoNature-atlas compatible with Debian 12.
We should now support Debian 11 and 12, and stop supporting Debian 9 and 10.
Another solution is to use the Docker that is available since GeoNature-atlas 1.6.

**Comment by rastaferraille** [comment]
Hello Camille,

Okay, thank you very much for your really quick answer (as usual). I try with docker !

**Comment by lpofredc** [comment]
It seems that a fix have been released for greenlet 1.1 on python 3.11: https://github.com/python-greenlet/greenlet/issues/305

Atlas uses greenlet 1.1.2, patched version is 1.1.3

**Comment by camillemonchicourt** [comment]
It should be easy to upgrade it in https://github.com/PnX-SI/GeoNature-atlas/blob/master/requirements.txt#L39 (which is automatically generated based on https://github.com/PnX-SI/GeoNature-atlas/blob/master/requirements.in.

**Comment by Jacobe2169** [comment]
@lpofredc Did the update of greenlet fix the setup  ?

**Comment by lpofredc** [comment]
Not tested on GeoNature-atlas but it did the trick on GeoNature-citizen

**Comment by rastaferraille** [comment]
I haven't got the error anymore with greenlet, thank you !

Now installation gives me this error :

La relation atlas.t_mailles_territoire n'existe pas

Idem pour vm_observations_mailles et vm_cor_taxon_organism.

Le lun. 9 oct. 2023 à 09:54, Jacques Fize ***@***.***> a
écrit :

**Comment by camillemonchicourt** [comment]
Autre soucis, à voir dans un autre ticket je pense.
La création de la BDD semble avoir eu un soucis.
Il faudrait tous les logs de l'installation pour identifier le soucis.
Ou relancer l'installation complète de la BDD.

**Comment by rastaferraille** [comment]
Ok, désolé, je vais tout relancer. Merci !

**Comment by jpm-cbna** [comment]
> @lpofredc @rastaferraille Did the update of greenlet fix the setup ?

Replace `greenlet==1.1.2` with `greenlet==1.1.3` in `requirements.txt` file to resolve the issue on Debian 12.1.

**Comment by camillemonchicourt** [comment]
La version de 1.7.0 de GN-atlas compatible avec Debian 11 et 12 ainsi qu'avec GeoNature 2.15 (et plus) a été publiée.
